### ISSUE TITLE
reify-fake doesn't support varargs interface methods

### ISSUE THREAD

**Issue by schmee** [metametadata/clj-fakes#4]
In the source, varargs methods are explicitly removed: https://github.com/metametadata/clj-fakes/blob/99785d6bbde1ff640eabd2b28a26996698896762/src/clj_fakes/context.cljc#L587 Is this a technical limitation, a design choice or something else entirely? 😄

**Comment by metametadata** [comment]
Frankly speaking, I don't remember exactly why I left it like this. But I've quickly tested the possible solution and it looks doable. Do you have a real use case for this feature already?

**Comment by schmee** [comment]
I wanted to use `reify-fake` and specifically `:recorded-fake` to write some tests involving a [WebSocket.Builder](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/WebSocket.Builder.html). Instead, I ended up hand-rolling the solution below, but this seems like exactly the kind of thing I don't want to write myself and instead use clj-fakes for 🙂 

```clj
(defn update-arg! [a k args]
  (swap! a update k (fnil conj []) args))

(defrecord FakeBuilder [args]
  WebSocket$Builder
  (header [this a b] (update-arg! args :header [a b]) this)
  (subprotocols [this a rs] (update-arg! args :subprotocols [a rs]) this)
  (connectTimeout [this t] (update-arg! args :connectTimeout [t]) this))
```

**Comment by metametadata** [comment]
Yeah, I see, thanks. I'll try to release it somewhere next month.

**Comment by metametadata** [comment]
Fixed in v0.12.0. Example usage: https://github.com/metametadata/clj-fakes/blob/d928ddfd11b150b1cd2df5621265c447bf42395a/test/unit/reify_fake.cljc#L476.
